sdks/python/README.md

```diff
@@ -2,18 +2,105 @@
 
 The Redactive Python SDK provides a robust and intuitive interface for interacting with the Redactive platform, enabling developers to seamlessly integrate powerful data redaction and anonymization capabilities into their Python applications.
 
-## Building the Python SDK
+## Installation
 
-`cd sdks/python`
+You don't need this source code unless you want to modify the package. If you just
+want to use the package, just run:
 
-`python -m pip install --upgrade build`
+```sh
+pip install --upgrade redactive
+```
 
-`python -m build`
+Install from source with:
 
-`python install -e .`
+```sh
+python -m pip install .
+```
 
-## Building the optional reranker
+### Requirements
 
-`cd sdks/python`
+- Python 3.11+
 
-`pip install -e ".[reranking]"`
+## Usage
+
+The library has following components.
+
+- **AuthClient** - provides functionality to interact with data sources
+- **SearchClient** - provides functionality to search chunks with Redactive search service in gRpc
+- **RerankingSearchClient** [Experimental] - retrieves extra results, then re-ranks them using a more precise ranking function, returning the top_k results
+
+### AuthClient
+
+AuthClient needs to be configured with your account's api key which is
+available in Apps page in [Redactive Dashboard](https://dashboard.redactive.ai/).
+
+```python
+from redactive.auth_client import AuthClient
+
+client = AuthClient(api_key="API-KEY")
+
+# Establish an connection to data source
+# Possible data sources: confluence, google-drive, jira, zendesk, slack, sharepoint
+redirect_uri = "https://url-debugger.vercel.app"
+sign_in_url = await client.begin_connection(
+    provider="confluence", redirect_uri=redirect_uri
+)
+
+# Navigate User to sign_in_url
+# User will receive an oauth2 auth code after consenting the app's data source access permissions.
+# Use this code to exchange Redactive access_token with Redactive API
+response = await client.exchange_tokens(code="OAUTH2-TOKEN")
+```
+
+### SearchClient
+
+With Redactive access_token, User can search documents with Redactive Search service.
+
+```python
+from redactive.search_client import SearchClient
+
+client = SearchClient()
+
+client.query_chunks(
+    access_token="REDACITVE-USER-ACCESS-TOKEN",
+    semantic_query="Tell me about AI"
+)
+```
+
+## Development
+
+Python SDK code is under `sdks/python` in Redactive Github Repository.
+
+To format code, run
```

Comment:
In order to comply with the repository style guide, we recommend running the following tools.

To format your code, run: